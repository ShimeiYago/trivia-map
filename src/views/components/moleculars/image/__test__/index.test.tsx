import { shallow, ShallowWrapper } from 'enzyme';
import { Image, Props } from '..';

let wrapper: ShallowWrapper<Props, unknown, Image>;

const props: Props = {
  src: 'image-data',
  alt: 'alt-text',
  onClick: jest.fn(),
};

describe('Shallow Snapshot Tests', () => {
  beforeEach(() => {
    wrapper = shallow(<Image {...props} />);
  });

  it('basic', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('width', () => {
    wrapper.setProps({
      width: 'full',
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('height', () => {
    wrapper.setProps({
      height: 'full',
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('objectFit cover', () => {
    wrapper.setProps({
      objectFit: 'cover',
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('border radius', () => {
    wrapper.setProps({
      borderRadius: true,
    });
    expect(wrapper).toMatchSnapshot();
  });
});
