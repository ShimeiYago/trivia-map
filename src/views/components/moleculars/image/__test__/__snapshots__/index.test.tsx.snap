// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Snapshot Tests basic 1`] = `
<Img
  alt="alt-text"
  className="image"
  loader={
    <div
      className="loader"
    >
      <CenterSpinner />
    </div>
  }
  onClick={[MockFunction]}
  src="image-data"
/>
`;

exports[`Shallow Snapshot Tests border radius 1`] = `
<Img
  alt="alt-text"
  className="image border-radius"
  loader={
    <div
      className="loader"
    >
      <CenterSpinner />
    </div>
  }
  onClick={[MockFunction]}
  src="image-data"
/>
`;

exports[`Shallow Snapshot Tests height 1`] = `
<Img
  alt="alt-text"
  className="image height-full"
  loader={
    <div
      className="loader"
    >
      <CenterSpinner />
    </div>
  }
  onClick={[MockFunction]}
  src="image-data"
/>
`;

exports[`Shallow Snapshot Tests objectFit cover 1`] = `
<Img
  alt="alt-text"
  className="image cover"
  loader={
    <div
      className="loader"
    >
      <CenterSpinner />
    </div>
  }
  onClick={[MockFunction]}
  src="image-data"
/>
`;

exports[`Shallow Snapshot Tests width 1`] = `
<Img
  alt="alt-text"
  className="image width-full"
  loader={
    <div
      className="loader"
    >
      <CenterSpinner />
    </div>
  }
  onClick={[MockFunction]}
  src="image-data"
/>
`;
